src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    gSaveContext.respawn[0].entranceIndex = globalCtx->nextEntranceIndex;
+    func_80169EFC(globalCtx); /* Load new entrance? */
+    gSaveContext.respawnFlag = -2;
+    this->actionFunc = EnBji01_DoNothing;
+}
+
+void EnBji01_DoNothing(EnBji01* this, GlobalContext* globalCtx) {
+}
+
+void func_809CD6C0(EnBji01* this, GlobalContext* globalCtx) {
+
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 2, &this->animationIndex);
+    this->actionFunc = func_809CD70C;
+}
+
+void func_809CD70C(EnBji01* this, GlobalContext* globalCtx) {
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x444);
+    func_809CCDE0(this, globalCtx);
+    if (this->actor.shape.rot.y == this->actor.yawTowardsPlayer) {
+        func_800B86C8(&this->moonsTear->actor, globalCtx, &this->actor); /* Z-Target the Moon's Tear? */
+        this->actionFunc = func_809CD77C;
+    }
+}
+
+void func_809CD77C(EnBji01* this, GlobalContext* globalCtx) {
+    if (this->moonsTear->actor.colChkInfo.health == 0) {
+        func_809CCE98(this, globalCtx);
+    }
+}
+
+void EnBji01_Init(Actor* thisx, GlobalContext* globalCtx) {
+
+    EnBji01* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 30.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600578C, &D_06000FDC, this->jointTable, this->morphTable, 0x10);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 0;
+    this->actor.child = NULL;
+    this->animationIndex = -1;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    func_8013E3B8(&this->actor, this->cutscenes, ARRAY_COUNT(this->cutscenes)); /* initializes all elements of cutscenes to -1 */
+    this->moonsTear =
+        (ObjMoonStone*)func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_PROP, ACTOR_OBJ_MOON_STONE);
+
+    switch (gSaveContext.entranceIndex) {
+        case 0x4C00: /* Observatory from ECT */
+        case 0x4C10: /* Observatory from Termina Field door */
+            this->actor.params = 0;
+            func_809CCE98(this, globalCtx);
+            break;
+        case 0x4C20: /* Observatory from Termina Field telescope */
+            this->actor.flags |= 0x10000;
+            func_801A5BD0(0);
+            func_801A89A8(0xE0000100);
+            this->actor.params = 3;
+            func_809CCE98(this, globalCtx);
+            break;
+        default:
+            Actor_MarkForDeath(&this->actor);
+            break;
+    }
+}
+
+void EnBji01_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+
+    EnBji01* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnBji01_Update(Actor* thisx, GlobalContext* globalCtx) {
+
+    static s16 sBlinkSequence[6] = { 0, 1, 2, 1, 0, 0 };
+    EnBji01* this = THIS;
+
+    s32 pad; /*pad until 4 main functions take gamestate that gets promoted to globalctx*/
```

Comment:
The comment isn't needed here, having `pad` all over the place is pretty common.
```suggestion
void EnBji01_Update(Actor* thisx, GlobalContext* globalCtx) {
    static s16 sBlinkSequence[] = { 0, 1, 2, 1, 0, 0 };
    EnBji01* this = THIS;
    s32 pad;
```